### ISSUE TITLE
Use Github container registry instead of Dockerhub

### ISSUE THREAD

**Issue by rikukissa** [opencrvs/opencrvs-core#6910]
### Description  

Migrate OpenCRVS Core to use [GitHub Container Registry (GHCR)](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry) instead of Docker Hub. This will reduce deployment costs (>1000 USD/year for Docker Hub). Countries will continue using Docker Hub for now.

### Dev tasks  

- [x] Familiarize yourself with [GitHub Container Registry (GHCR)](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry).  
- [x] Update OpenCRVS Core’s GitHub Actions workflows to push images to GHCR instead of Docker Hub.  
- [x] Update OpenCRVS services to pull images from GHCR.  
- [x] Update country config repository to fetch core images from GHCR by default.  
- [x] Deploy to a new environment using GHCR-hosted images.  
- [x] Make a pull request with the updated CI/CD configuration for OpenCRVS Core.  
- [x] Update documentation in the [documentation repository](https://github.com/opencrvs/documentation) to replace Docker Hub references with GHCR.

**Comment by n1koo** [comment]
In addition to the technical bits we also have to double check the pricing implications of this. Theres novel differences between how Github charges for network traffic that might play an impact.

Also on the techical side - we probably should also double check the PoPs of this Github's solution vs Docker Hub. Eg. just to double check the connectivity is there for developing countries and not only US/EU centric

**Comment by adskyiproger** [comment]
Github allows only 500M free space for docker images (or any other packages) for private repositories:
https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-packages/about-billing-for-github-packages

Even paid plans are very limited, it makes it useless for Country configs

**Comment by rikukissa** [comment]
Indeed, even the **GitHub Pro | 2GB** tier wouldn't be enough for countries as in active development, there can be tens if not hundreds of images each 500MB or more. Let's park this change for now

**Comment by rikukissa** [comment]
Reopened. Let's do this for all public OpenCRVS images except country config and Farajaland

**Comment by adskyiproger** [comment]
Documentation update required:
- https://documentation.opencrvs.org/general/releases#opencrvs-gitflow-and-quality-gates
   - Check diagrams: there are references to dockerhub, but dockerhub will be used by country config only
- https://documentation.opencrvs.org/setup/3.-installation/3.3-set-up-a-server-hosted-environment/4.3.4-create-a-github-environment/4.3.4.1-environment-secrets-and-variables-explained:
   - Describe explicitly: Core is stored on GitHub, Countryconfig is stored on Dockerhub
   - Add manual page that explain how to change container registry
